packages/browser-pool/src/browser-pool.ts

```diff
@@ -451,7 +451,7 @@ export class BrowserPool<
                 browserController = await this._launchBrowser(id, { browserPlugin, proxyTier, proxyUrl });
             tryCancel();
 
-            return this._createPageForBrowser(id, browserController, pageOptions, proxyUrl);
+            return await this._createPageForBrowser(id, browserController, pageOptions, proxyUrl);
```

Comment:
and i am also not convinced this can help, the inner function was also async, returning from inside an async function is enough.